build.cmd

```diff
@@ -1,2 +1,3 @@
+git submodule update --init
```

Comment:
This shouldn't be inside of our build scripts. This means this will run every time someone tries to build the repo.

If this is needed for CI, then this step should be its own step inside of CI.

Reply:
Could you tell me the exact location in your mind?

Reply:
For PR validation, you can add a new step above this line:

https://github.com/dotnet/machinelearning/blob/53cdb20358e7ae98cdaf6de198e35f48fbdac269/build/ci/phase-template.yml#L20

For the official build there are 3 places needed:

https://github.com/dotnet/machinelearning/blob/53cdb20358e7ae98cdaf6de198e35f48fbdac269/build/vsts-ci.yml#L26

https://github.com/dotnet/machinelearning/blob/53cdb20358e7ae98cdaf6de198e35f48fbdac269/build/vsts-ci.yml#L50

https://github.com/dotnet/machinelearning/blob/53cdb20358e7ae98cdaf6de198e35f48fbdac269/build/vsts-ci.yml#L89

Although, my preference would be to get it working with VSTS's built-in support: https://docs.microsoft.com/en-us/azure/devops/pipelines/build/repository?view=vsts#checkout-submodules.

Or, since we are only using a single file, maybe using submodules is too heavy of a hand here. The CoreCLR didn't use a submodule in order to use/build libunwind into .NET Core - https://github.com/dotnet/coreclr/pull/17094. Instead, here are hundreds of files that were copied into dotnet/coreclr.

Reply:
Let my try your idea.

For build-in support, I don't think it would work because we have tried it in the beginning. My submodule is a public repo, but Azure just didn't clone it.

Submodule was the conclusion obtained from several discussion including a scrum. Please allow me to stay with it.

Reply:
Your idea works so the build script doesn't initialize submodules anymore. Thanks.